homeassistant/components/smartthings/climate.py

```diff
@@ -338,6 +366,9 @@ async def async_set_fan_mode(self, fan_mode: str) -> None:
         # the entity state ahead of receiving the confirming push updates
         self.async_write_ha_state()
 
+        # setting the fan must reset the preset mode (it deactivates the windFree function)
+        self._attr_preset_mode = ""
```

Comment:
This has to go before writing the state or it will not apply until next update cycle
Same for places below.

Reply:
i see, thanks